### PR TITLE
devops cleanups

### PR DESCRIPTION
**Pull request by astelmach01**
- Added pip-tools functionality. Instead of adding to requirements.txt, simply add the library to requirements.in and run pip-compile after installing pip-tools
- Cleaned up dockerfile to install requirements first for a quick optimization
- Changed docker image from 3.8 -> 3.10
- Added a proper gitignore

### CODE REVIEW

**Comment by ryanpeach** [comment]
Also related https://github.com/Torantulino/Auto-GPT/issues/137

File: Dockerfile [hunk 1]
```diff
@@ -1,8 +1,12 @@
-FROM python:3.11
+FROM python:3.10-slim
```
Comment (nponeccop) [inline Dockerfile:1]: Python bump is and using a slim image are separate design decisions. Use 2 separate PRs for this single line as it's easier to approve and merge them.

File: Dockerfile [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
 
 WORKDIR /app
-COPY scripts/ /app
 
-RUN pip install -r requirements.txt
+COPY requirements.txt /app/requirements.txt
```
Comment (nponeccop) [inline Dockerfile:5]: This change is done elsewhere (in different PRs) already. Search for the PR number(s) and add a comment. Remove from this PR.

File: requirements.in
Comment (nponeccop) [inline requirements.in:1]: It's a good idea to generate requirements.txt for a more precice dependency pindown. Explain the advantages and submit only this improvement as a separate PR.

File: scripts/execute_code.py [hunk 3]
```diff
@@ -18,7 +18,7 @@ def execute_python_file(file):
     try:
         client = docker.from_env()
 
-        # You can replace 'python:3.8' with the desired Python image/version
```
Comment (nponeccop) [inline scripts/execute_code.py:21]: This typo should be a separate PR independent of all the above.

**Review by nponeccop — CHANGES_REQUESTED** [review]
Split into independent atomic PRs. While I'm not a maintainer, the current maintainers have a similar requirement in other PRs.

**Pull request closed without merging**